### ISSUE TITLE
StackOverflow error due to indirect recursion

### ISSUE THREAD

**Issue by guilhermejcgois** [FasterXML/jackson-databind#1280]
I get this error:

```
java.lang.StackOverflowError: null
	at java.lang.ClassLoader.defineClass1(Native Method) ~[na:1.8.0_73]
	at java.lang.ClassLoader.defineClass(ClassLoader.java:760) ~[na:1.8.0_73]
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142) ~[na:1.8.0_73]
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467) ~[na:1.8.0_73]
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73) ~[na:1.8.0_73]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368) ~[na:1.8.0_73]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362) ~[na:1.8.0_73]
	at java.security.AccessController.doPrivileged(Native Method) ~[na:1.8.0_73]
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361) ~[na:1.8.0_73]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[na:1.8.0_73]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331) ~[na:1.8.0_73]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[na:1.8.0_73]
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:691) ~[jackson-databind-2.6.6.jar:2.6.6]
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:157) ~[jackson-databind-2.6.6.jar:2.6.6]
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:693) ~[jackson-databind-2.6.6.jar:2.6.6]
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:675) ~[jackson-databind-2.6.6.jar:2.6.6]
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:157) ~[jackson-databind-2.6.6.jar:2.6.6]
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:693) ~[jackson-databind-2.6.6.jar:2.6.6]
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:675) ~[jackson-databind-2.6.6.jar:2.6.6]
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:157) ~[jackson-databind-2.6.6.jar:2.6.6]
```

I have the two class in my JPA library:
```
package model;
...
public class A { 
	private Long id;
	...
	public AEntity getEntity() {
		AEntity entity = new AEntity();
		entity.setId(id);
		...
		return entity;
	}
}
```

```
package entity;
...
public class AEntity {
	@Id
	private Long id;
	...
	public A getModel() {
		A model = new A();
		model.setId(id);
		...
		return model;
	}
}
```
So, when I try to get a JSON representation of A, I got the above error. If I remove A::getEntity or AEntity::getModel, it's all ok. But I don't want to do this.

Can someone give some light or help? Could this be a bug? Or a bad design in my library?

Thanks for now! :grinning:

**Comment by cowtowncoder** [comment]
Please send usage questions on `jackson-user` list (see https://groups.google.com/forum/#!forum/jackson-user); issue tracker is for reporting issues. Mailing list has more active users, beyond core development team, and users are often better at suggesting work-arounds or design improvements.

Exception itself is not enough to say much unfortunately, so it will make sense to add more code to show usage. It is also good to mention Jackson version in use (and obviously try to use a more recent version, if one available -- 2.7.5 is the latest stable).

**Comment by guilhermejcgois** [comment]
Thanks for the feedback, @cowtowncoder.
